core/src/components/TextArea/Markdown.tsx

```diff
@@ -26,30 +20,43 @@ export default function Markdown(props: MarkdownProps) {
   const { prefixCls } = props;
   const { markdown = '', highlightEnable, dispatch } = useContext(EditorContext);
   const preRef = React.createRef<HTMLPreElement>();
+  const [mdStr, setMdStr] = useState('');
+
   useEffect(() => {
     if (preRef.current && dispatch) {
       dispatch({ textareaPre: preRef.current });
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
-  if (!markdown) {
-    return <pre ref={preRef} className={`${prefixCls}-text-pre wmde-markdown-color`} />;
-  }
-  let mdStr = `<pre class="language-markdown ${prefixCls}-text-pre wmde-markdown-color"><code class="language-markdown">${html2Escape(
-    String.raw`${markdown}`,
-  )}\n</code></pre>`;
-
-  if (highlightEnable) {
-    try {
-      mdStr = rehype()
-        .data('settings', { fragment: true })
-        // https://github.com/uiwjs/react-md-editor/issues/593
-        // @ts-ignore
-        .use(rehypePrism, { ignoreMissing: true })
-        .processSync(mdStr)
-        .toString();
-    } catch (error) {}
-  }
+
+  useEffect(() => {
+    async function processMarkdown() {
+      if (!markdown) {
+        setMdStr('');
+        return;
+      }
+
+      let mdStr = `<pre class="language-markdown ${prefixCls}-text-pre wmde-markdown-color"><code class="language-markdown">${html2Escape(
+        String.raw`${markdown}`,
+      )}\n</code></pre>`;
+
+      if (highlightEnable) {
+        try {
+          mdStr = await rehype()
+            .data('settings', { fragment: true })
+            // https://github.com/uiwjs/react-md-editor/issues/593
+            // @ts-ignore
+            .use(rehypePrism, { ignoreMissing: true })
+            .process(mdStr)
```

Comment:
<img width="444" alt="image" src="https://github.com/user-attachments/assets/d5ffde87-fb0e-4ed3-bb5e-2b014e081d22">

An error occurred in the example. @vishwamartur